### PR TITLE
Feature implementation: noLights()

### PR DESCRIPTION
**Pull request by sanketsingh24**
Fixes: #3945 

Work done- 
- [x] Implemented `noLights()` in /src/webgl/light.js
- [x] Added a manual-test-example in `/webgl/lights/noLights/`
- [x] Unit tests
- [x] Example
- [x] Documentation

To do-

### CODE REVIEW

File: src/webgl/light.js [hunk 1]
```diff
@@ -832,4 +832,64 @@ p5.prototype.spotLight = function(
...
+ *   createCanvas(100, 100, WEBGL);
+ * }
+ * function draw() {
+ *   background(0);
+ *   noStroke();
+ *   ambientLight(150, 0, 0);
+ *   translate(-25, 0, 0);
+ *   ambientMaterial(250);
+ *   sphere(20);
+ *   noLights();
```
Comment (aferriss) [inline src/webgl/light.js]: Can we put a line break here? (above noLights()) to separate the two sections

File: test/manual-test-examples/webgl/lights/noLights/sketch.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+function draw() {
+  background(0);
+
+  ambientLight(150, 0, 0);
+  translate(-200, 0, 0);
+  ambientMaterial(250);
+  sphere(50, 64);
+  noLights();
```
Comment (aferriss) [inline test/manual-test-examples/webgl/lights/noLights/sketch.js]: Line break here as well please!

File: src/webgl/light.js [hunk 3]
```diff
@@ -832,4 +832,64 @@ p5.prototype.spotLight = function(
...
+ *   ambientLight(0, 150, 0);
+ *   translate(50, 0, 0);
+ *   ambientMaterial(250);
+ *   sphere(20);
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * Something
```
Comment (aferriss) [inline src/webgl/light.js]: Make sure we update the alt tag here to reflect the example.

File: src/webgl/light.js [hunk 4]
```diff
@@ -832,4 +832,64 @@ p5.prototype.spotLight = function(
   return this;
 };
 
+/**
+ * This function will remove all the lights from the sketch for the
+ * subsequent materials rendered. It affects all the subsequent methods.
```
Comment (aferriss) [inline src/webgl/light.js]: Can we put a note in here about how to re-enable lights?

Comment (sanketsingh24) [reply src/webgl/light.js]: Something like this: `To Re-enable lights, you need to call the lighting methods again`

Comment (aferriss) [reply src/webgl/light.js]: How about "Calls to lighting methods made after `noLights()` will re-enable lights in the scene."

**Pull request merged**